app/Models/Actionlog.php

```diff
@@ -91,15 +90,14 @@ public function parentlog()
     /**
        * Check if the file exists, and if it does, force a download
        **/
-    public function get_src($type = 'assets')
+    public function get_src($type = 'assets', $fieldname = 'filename')
```

Comment:
This seemed like a strange place for this method to exist to me when I was polymorphicating everything, but I was afraid to touch it... also unclear exactly what the method does when looking at it quickly.  Does associating files with a log in general make sense? (The signatures do, wondering about the asset files)--Not really related to your sig, just a ponder I had when staring at all this code a few weeks ago.